src/utilities/sequence-editor/sequence-linter.ts

```diff
@@ -302,141 +304,163 @@ function validateCommandTypeMixing(node: SyntaxNode): Diagnostic[] {
   return diagnostics;
 }
 
-function validateLocals(locals: SyntaxNode[], text: string) {
+function validateVariables(inputParams: SyntaxNode[], text: string, type: 'INPUT_PARAMS' | 'LOCALS' = 'LOCALS') {
   const variables: VariableDeclaration[] = [];
   const diagnostics: Diagnostic[] = [];
+
+  if (inputParams.length === 0) {
+    return {
+      diagnostics,
+      variables,
+    };
+  }
+
   diagnostics.push(
-    ...locals.slice(1).map(
-      local =>
+    ...inputParams.slice(1).map(
+      inputParam =>
         ({
-          ...getFromAndTo([local]),
-          message: 'There is a maximum of one @LOCALS directive per sequence',
+          ...getFromAndTo([inputParam]),
+          message: `There is a maximum of ${type} directive per sequence`,
           severity: 'error',
         }) as const,
     ),
   );
-  locals.forEach(local => {
-    let child = local.firstChild;
-    while (child) {
-      if (child.name !== 'Enum') {
+
+  if (inputParams[0].getChildren('Variable').length === 0) {
+    diagnostics.push({
+      from: inputParams[0].from,
+      message: `Missing values for ${type} directive`,
+      severity: 'error',
+      to: inputParams[0].to,
+    });
+  }
+
+  inputParams[0].getChildren('Variable').forEach(parameter => {
+    const typeNode = parameter.getChild('Type');
+    const enumNode = parameter.getChild('EnumName');
+    const rangeNode = parameter.getChild('Range');
+    const objectNode = parameter.getChild('Object');
+
+    const { enumName, name, range, type } = getVariableInfo(parameter, text);
+
+    if (type) {
+      if (['FLOAT', 'INT', 'STRING', 'UINT', 'ENUM'].includes(type) === false) {
         diagnostics.push({
-          from: child.from,
-          message: `@LOCALS values are required to be Enums`,
+          from: typeNode?.from ?? objectNode?.from ?? parameter.from,
```

Comment:
maybe rewrite as below to capture that 's the same node we're pulling from and to attribute off of
```
const node = typeNode ?? objectNode ?? parameter;
const { from, to } = node;
```

Reply:
Addressed with commit 3244803be50f2ee2de696c1459c4dc03d3d4bb87